### PR TITLE
Add Documentation for .NET SDK Container Build

### PR DESCRIPTION
**Pull request by alexyaang**

### CODE REVIEW

**Comment by alexyaang** [comment]
@ghogen @bwateratmsft @danegsta

Let me know your thoughts!

File: docs/containers/debug-netcore.md [hunk 1]
```diff
@@ -19,8 +19,28 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
...
-1. Open the project folder in VS Code.
+- If needed, create a .NET project with `dotnet new`.
+- Open the project folder in VS Code.
+- You are prompted with whether to build your image using a Dockerfile or the [.NET SDK](https://www.microsoft.com/net/download). If you want a simple container debug experience without a Dockerfile or if you are more familiar with MSBuild, choose **Use .NET SDK**. If you're more comfortable working with Dockerfiles, choose **Use a Dockerfile**.
+
+### .NET SDK Container Build (Debug without `Dockerfile`)
+
+// TODO: add context
+
+1. Press `kb(workbench.action.debug.start)` or choose **Start Debugging** from the **Run** menu.
```
Comment (bwateratmsft) [inline docs/containers/debug-netcore.md:30]: Does `kb(workbench.action.debug.start)` render prettier on the docs site?

File: docs/containers/images/debug/debugger-scaffolding.png
Comment (bwateratmsft) [inline docs/containers/images/debug/debugger-scaffolding.png:1]: This image seems like it's off center or over-cropped. I'd like to see the tooltip that says "Open 'launch.json'". You can also cut some of the empty space below. @ghogen might also have thoughts on it.

File: docs/containers/debug-netcore.md [hunk 2]
```diff
@@ -60,3 +80,18 @@ To enable SSL (using the HTTPS protocol), you will need to make a few changes to
...
+## Saving Project File Preference for [.NET SDK Container Build](https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container)
+
+If you have a workspace folder containing multiple .NET project files and you only intend to debug one for the foreseeable future (i.e. you never want to be prompted a list of project files when you `kb(workbench.action.debug.start)`). You can save your launch profile by doing the following:
+
+1. Follow the steps in [.NET SDK Container Build](#net-sdk-container-build-debug-without-dockerfile) and keep the debug session live
+1. Click on the `gear` icon in your debugger view.
+
+   ![dockerSharedFolders](images/debug/debugger-scaffolding.png)
+
+1. Select **Docker: Add Docker Files to Workspace...**
```
Comment (bwateratmsft) [inline docs/containers/debug-netcore.md:93]: Wait, can't they scaffold the new debug launch configuration and specify a project file in it, and not have to add a Dockerfile?

File: docs/containers/debug-netcore.md (hunk 1 above)
Comment (gregvanl) [reply docs/containers/debug-netcore.md:30]: Yep, this renders as the correct keyboard shortcut for the readers platform, with a hover for the other two supported.

image

Not so interesting for `F5` but nice for Ctrl+* on Windows/Linux versus Cmd+* on macOS

Comment (alexyaang) [reply docs/containers/debug-netcore.md:30]: Nice!

File: docs/containers/debug-netcore.md (hunk 2 above)
Comment (alexyaang) [reply docs/containers/debug-netcore.md:93]: ahhh shoot... clutch catch! I intended to put `Docker: Debug in Container` and not `Docker: Add Docker Files to Workspace...`

File: docs/containers/debug-netcore.md [hunk 3]
```diff
@@ -1,4 +1,3 @@
----
```
Comment (gregvanl) [inline docs/containers/debug-netcore.md:1]: Revert this change as the metadata header needs to be inside a `---` block.

**Comment by gregvanl** [comment]
Hi @alexyaang This PR is committing a new image directly to the vscode-docs repo as a binary file. To keep the repo size manageable, vscode-docs uses Git LFS for image and video files.
You can read how to install and enable Git LFS in the repo README at https://github.com/microsoft/vscode-docs#cloning
Can you redo this PR from a local repo with Git LFS enabled? Thanks

File: docs/containers/debug-common.md [hunk 4]
```diff
@@ -76,9 +75,16 @@ Example `launch.json` configuration for debugging a Python application:
 
 ## .NET
 
-More information about debugging .NET applications within Docker containers can be found in [Debug .NET within Docker containers](/docs/containers/debug-netcore.md).
+We offer two ways of building & debugging your project within Docker containers:
 
-Example `launch.json` configuration for debugging a .NET application:
+- **[With .NET SDK](/docs/containers/debug-netcore.md#net-sdk-container-build-debug-without-dockerfile)**: This option offers a simplified container debugging experience without the need for a `Dockerfile`. If you are familiar with `MSBuild` or want a straightforward approach, this is the recommended choice.
+  > Note: This option is only available for .NET SDK 7 and above and uses the `dotnet publish` command to build the image.
```
Comment (ghogen) [inline docs/containers/debug-common.md:81]: There's a specific syntax for Notes in VS Code docs. This doesn't follow it exactly. It might need to be separated by a blank line, indented to align with the previous paragraph, and use bold (**Note**:) but let's find other Notes in the docs and follow the pattern.

File: docs/containers/debug-common.md [hunk 5]
```diff
@@ -76,9 +75,16 @@ Example `launch.json` configuration for debugging a Python application:
...
+
+- **[With a Dockerfile](/docs/containers/debug-netcore.md#debug-with-dockerfile)**: If you prefer working / customizing your project with a `Dockerfile`, choose this option.
```
[the remark this answers is not in the document]
Comment (ghogen) [reply docs/containers/debug-common.md:83]: Avoid slashes in text

File: docs/containers/debug-netcore.md [hunk 6]
```diff
@@ -19,16 +18,35 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
...
+- Optionally, set a breakpoint.
+
+## .NET SDK vs. Dockerfile Build
+
+For those seeking a streamlined container debugging experience without the need for a Dockerfile or who are well-acquainted with MSBuild, the .NET SDK option is the ideal choice. On the other hand, if you feel more at ease working with Dockerfiles, opt for Dockerfile.
+
+### .NET SDK Container Build (Debug without `Dockerfile`)
+
+1. Press `kb(workbench.action.debug.start)` or choose **Start Debugging** from the **Run** menu. (Make sure you don't have any existing launch profiles in `launch.json`)
+1. You're going to be prompted with a list of debuggers, choose Select **Docker: Debug in Container**
```
Comment (ghogen) [inline docs/containers/debug-netcore.md:32]:
```suggestion
1. You're prompted with a list of debuggers. Choose **Docker: Debug in Container**
```

Comment (ghogen) [reply docs/containers/debug-netcore.md:32]: Use present tense. Also is the "Select" part of the UI or was it redundant with "choose" ?

File: docs/containers/debug-netcore.md [hunk 7]
```diff
@@ -19,16 +18,35 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
...
+1. You will be prompted with options to either build with a `Dockerfile` (**Use a Dockerfile**) or build using the .NET SDK (**Use .NET SDK**)
```
Comment (ghogen) [inline docs/containers/debug-netcore.md:33]:
```suggestion
1. When prompted with options to either build with a `Dockerfile` (**Use a Dockerfile**) or build using the .NET SDK (**Use .NET SDK**), select **Use .NET SDK**.
```
Combining these two steps, since otherwise this step doesn't have a user action.

File: docs/containers/debug-netcore.md [hunk 8]
```diff
@@ -19,16 +18,35 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
...
+1. Select **Use .NET SDK**
```
Comment (ghogen) [inline docs/containers/debug-netcore.md:34]:
```suggestion
```

File: docs/containers/debug-netcore.md [hunk 9]
```diff
@@ -19,16 +18,35 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
...
+1. If you have multiple project files in your workspace, choose the project file associated with the project you want to debug
```
Comment (ghogen) [inline docs/containers/debug-netcore.md:35]:
```suggestion
1. If you have multiple project files in your workspace, choose the project file associated with the project you want to debug.
```

File: docs/containers/debug-netcore.md [hunk 10]
```diff
@@ -19,16 +18,35 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
...
+1. Your .NET app will run in a Docker container, and the web app will open in your browser.
+
+> Note: **Supported .NET SDK Versions:** This feature is available for .NET SDK version 7.0.300 and above by default. For versions between 7.0.100 and 7.0.300, enable it with `dotnet add package Microsoft.NET.Build.Containers`. You can read more about .NET SDK Container build on [Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container).
```
Comment (ghogen) [inline docs/containers/debug-netcore.md:38]: Follow the official Note syntax

File: docs/containers/debug-netcore.md [hunk 11]
```diff
@@ -60,3 +78,16 @@ To enable SSL (using the HTTPS protocol), you will need to make a few changes to
    ```
 
 For additional customization options, see the documentation on [Tasks](/docs/containers/reference.md) and [Debug containerized apps](/docs/containers/debug-common.md).
+
+## Saving Project File Preference for [.NET SDK Container Build](https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container)
```
Comment (ghogen) [inline docs/containers/debug-netcore.md:82]: I doubt if links are allowed in headings

File: docs/containers/debug-netcore.md [hunk 12]
```diff
@@ -60,3 +78,16 @@ To enable SSL (using the HTTPS protocol), you will need to make a few changes to
...
+
+If you have a workspace folder with multiple .NET project files and you want to exclusively debug one specific project (without being prompted to choose from a list of project files every time you `kb(workbench.action.debug.start)`), you can save your launch profile by following these steps:
+
+1. Follow the steps in [.NET SDK Container Build](#net-sdk-container-build-debug-without-dockerfile) and keep the debug session live
```
[the remark this answers is not in the document]
Comment (ghogen) [reply docs/containers/debug-netcore.md:86]: Periods at the end of full sentences when they appear in lists.

File: docs/containers/debug-netcore.md [hunk 13]
```diff
@@ -60,3 +78,16 @@ To enable SSL (using the HTTPS protocol), you will need to make a few changes to
...
+1. Click on the `gear` icon in your debugger view.
+
+   ![dockerSharedFolders](images/debug/debugger-scaffolding.png)
+
+1. Select **Docker: Debug in Container**
+1. Choose the project file associated with the project you want to debug
+1. Your project preference is saved & you no longer need to choose a project file on `kb(workbench.action.debug.start)`
```
Comment (ghogen) [inline docs/containers/debug-netcore.md:93]:
```suggestion
    Your project preference is saved, and you no longer need to choose a project file on `kb(workbench.action.debug.start)`
```

Comment (ghogen) [reply docs/containers/debug-netcore.md:93]: Each step should be a user action. For other text, you can just indent it to align with the start of the previous paragraph (doublecheck the indentation)

File: docs/containers/debug-netcore.md [hunk 14]
```diff
@@ -9,7 +8,7 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
 
 ## Prerequisites
 
-1. Install the [.NET SDK](https://www.microsoft.com/net/download), which includes support for attaching to the .NET debugger.
+1. Install the [.NET SDK](https://www.microsoft.com/net/download), which includes support for attaching to the .NET debugger. If you install .NET SDK 7 or later, you have the option of debugging without a Dockerfile.
```
Comment (ghogen) [inline docs/containers/debug-netcore.md:11]:
```suggestion
1. Install the [.NET SDK](https://www.microsoft.com/net/download), which includes support for attaching to the .NET debugger. With .NET SDK 7 or later, you have the option of debugging without a Dockerfile.
```

Comment (ghogen) [reply docs/containers/debug-netcore.md:11]: I suggest this shorter wording which also removes any mention of installing, which might not be relevant to the user (for example if they're using a machine with different frameworks already installed).

File: docs/containers/debug-common.md (hunk 4 above)
Comment (alexyaang) [reply docs/containers/debug-common.md:81]: sounds good

File: docs/containers/debug-netcore.md [hunk 15]
```diff
@@ -19,16 +18,35 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
...
 
-1. If needed, create a .NET project with `dotnet new`.
-1. Open the project folder in VS Code.
+- If needed, create a .NET project with `dotnet new`.
+- Open the project folder in VS Code.
+- Optionally, set a breakpoint.
+
+## .NET SDK vs. Dockerfile Build
+
+For those seeking a streamlined container debugging experience without the need for a Dockerfile or who are well-acquainted with MSBuild, the .NET SDK option is the ideal choice. On the other hand, if you feel more at ease working with Dockerfiles, opt for Dockerfile.
```
[the remark this answers is not in the document]
Comment (alexyaang) [reply docs/containers/debug-netcore.md:27]: yeah that makes sense

File: docs/containers/debug-netcore.md (hunk 6 above)
Comment (alexyaang) [reply docs/containers/debug-netcore.md:32]: Good catch, "Select" is indeed redundant

File: docs/containers/debug-netcore.md (hunk 7 above)
Comment (alexyaang) [reply docs/containers/debug-netcore.md:33]: I like it

File: docs/containers/debug-netcore.md (hunk 13 above)
Comment (alexyaang) [reply docs/containers/debug-netcore.md:93]: Makes sense

File: docs/containers/debug-netcore.md [hunk 16]
```diff
@@ -19,16 +18,35 @@ MetaDescription: Debug a .NET app running in a Docker container, using Visual St
...
+
+### .NET SDK Container Build (Debug without `Dockerfile`)
+
+1. Press `kb(workbench.action.debug.start)` or choose **Start Debugging** from the **Run** menu. (Make sure you don't have any existing launch profiles in `launch.json`)
```
[the remark this answers is not in the document]
Comment (bwateratmsft) [reply docs/containers/debug-netcore.md:31]: I'm not really sure. I doubt VSCode themselves have any good way of addressing this...

[the remark this answers is not in the document]
Comment (alexyaang) [reply docs/containers/debug-netcore.md:31]: We could add a command to scaffold a .NET SDK Container build launch profile but... that sorta defies the intensions of this feature

**Comment by alexyaang** [comment]
Closing since I did not use Git LFS to commit the image. Opened a new one here https://github.com/microsoft/vscode-docs/pull/6544
